authors/apps/articles/views.py

```diff
@@ -0,0 +1,104 @@
+    """creating, viewing , deleting and updating articles"""
+
+    permission_classes = (IsAuthenticated,)
+
+
+    def post(self, request):
+        serializer_context = {'author': request.user.profile}
+        article = request.data.get('article', {})
+
+        # Create an article from the above data
+        serializer = ArticleSerializer(data=article, context=serializer_context)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response({"articles": serializer.data}, status=status.HTTP_201_CREATED)
+
+
+    def get(self, request):
+        queryset = Article.objects.all()
+        # the many param informs the serializer that it will be serializing more than a single article.
+        serializer = ArticleSerializer(queryset, many=True)
+        return Response({"articles": serializer.data})
+
+class ArticleRetrieveUpdateDelete(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ArticleSerializer
+    queryset = Article.objects.all()
+    lookup_field = 'slug'
+
+
+    def retrieve(self, request, slug):
+        try:
+            serializer_instance = self.queryset.get(slug=slug)
+        except Article.DoesNotExist:
+            raise NotFound('No article found with this slug.')
+
+        serializer = self.serializer_class(serializer_instance)
+        return Response({"articles": serializer.data})
+
+
+    def destroy(self, request, slug):
+        try:
+            serializer_instance = self.queryset.get(slug=slug)
+            requester = Profile.objects.get(user=request.user)
+            is_author = serializer_instance.author == requester
+            if not is_author:
+                return Response({"response": "you can't delete this article"},
+                            status=status.HTTP_403_FORBIDDEN)
+        except Article.DoesNotExist:
+            raise NotFound('No article found with this slug.')
+
+        self.perform_destroy(serializer_instance)
+        return Response({"message":"Article has been deleted"},
+            status=status.HTTP_204_NO_CONTENT)
+
+
+    def update(self, request, slug):
+        try:
+            serializer_instance = self.queryset.get(slug=slug)
+            requester = Profile.objects.get(user=request.user)
+            is_author = serializer_instance.author == requester
+            if not is_author:
+                return Response({"response": "you can't update this article"},
+                            status=status.HTTP_403_FORBIDDEN)
+        except Article.DoesNotExist:
+            raise NotFound('No article found with this slug.')
+
+        serializer_data = request.data.get('article', {})
+        serializer = self.serializer_class(
+            serializer_instance, data=serializer_data, partial=True
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response({"articles": serializer.data}, status=status.HTTP_202_ACCEPTED)
```

Comment:
@frankopkusianwar , instead of creating a response json, add a `renderer class` to the view